src/main/java/com/google/sps/servlets/PreviousQuestionServlet.java

```diff
@@ -0,0 +1,63 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import java.util.List; 
+
+import java.io.IOException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.cloud.firestore.Firestore;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentSnapshot;
+
+import com.google.sps.daos.GameInstanceDao;
+import com.google.sps.data.GameInstance;
+
+import java.util.List; 
+import java.util.ArrayList; 
+
+@WebServlet("/previousQuestion")
+public class PreviousQuestionServlet extends HttpServlet {
```

Comment:
Seems like there are pretty similar functionalities in next and previous at least, maybe even similar workflows otherwise. Can you extract some of this logic to a helper class?